### DISCUSSION TITLE
Finding the nearest config file(s)

### DISCUSSION THREAD

**Discussion by mrmckeb** [eslint/eslint#16972] [Q&A] [answered]
Hi all, we've used [`calculateConfigForFile`](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintcalculateconfigforfilefilepath) a few times and it works great.

We're working on something right now where we want to know where the nearest config file is, but we're not able to find a method that provides which config files are being resolved.

It looks like this happens here, but none of the related code is exposed hat we can see:
https://github.com/eslint/eslint/blob/c89a485c49450532ee3db74f2638429f1f37d0dd/lib/cli-engine/cli-engine.js#L957-L972

**Comment by nzakas** [comment]
Yes, we don't expose that information through the API, as it shouldn't matter to you as a consumer. 

We are also in the middle of switching config systems, so I'd advice caution in building anything on the eslintrc config system right now.

If you can explain more about your use case, we can discuss how you might do it with flat config.

**Reply by mrmckeb** [reply]
Thanks @nzakas, we're building some internal tooling and wanted to understand where the config file was located for a file or set of files. We could, as an example, then modify that config and turn on/off a rule.

I get that with config merging, it's also going to be hard to know which config has set a rule.

**Reply by nzakas** [reply]
That's interesting. I'm curious if you actually need to know the location of the config file for this? If you ultimately know what you want, couldn't you use the `overrideConfig` option to just modify the final output?

**Reply by mrmckeb** [reply]
We could do that, but I think we'd then need to set that up in their IDE, etc. There are some other use cases that we have for knowing the location of the file too, another example might be to provide a report on a repo's tooling and configs.

All of these have workarounds and we understand this is a really rare scenario - so we completely understand if you don't plan to expose that sort of information. I've been keeping an eye on the great work on flat config and the RFC on rewriting, and I understand all of that must be super time/brain consuming.

**Reply by nzakas** [reply]
Would it be helpful to add a `FlatESLint#findConfigFile()` method? With flat config, there is only ever one config file, so "nearest" becomes irrelevant.

**Reply by mrmckeb** [reply]
Yes, I think that would be great if you feel you can support that additional method!

**Reply by nzakas** [reply]
That seems like a reasonable addition to me. Can you open an issue requesting it? That way we can get team feedback and push this through the approval process.

**Reply by mrmckeb** [reply]
Done, thanks again!
https://github.com/eslint/eslint/issues/17046

**Comment by mrmckeb** [comment, marked as answer]
Closing this off in favour of the issue.

#17046
